### ISSUE TITLE
Elyra pipelines failing with version conflicts even with system owned images

### ISSUE THREAD

**Issue by yarnabrina** [elyra-ai/elyra#2719]
**Describe the issue**
`Elyra` is failing with a conflict, likely due to [requirements file](https://raw.githubusercontent.com/elyra-ai/elyra/v3.8.1/etc/generic/requirements-elyra.txt)

**To Reproduce**
1. Create a notebook with a single cell containing `print("hello world")`
2. Create a pipeline with image being `Pandas 1.4.1`
3. Submit a `Kubeflow` pipeline

**Screenshots or log output**
Error Message:

> ERROR: Cannot install nbclient==0.4.1 and nbconvert==6.4.5 because these package versions have conflicting dependencies.

<details>
<summary>Full Logs</summary>

```
Downloading https://raw.githubusercontent.com/elyra-ai/elyra/v3.8.1/elyra/kfp/bootstrapper.py
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 26 28961   26  7748    0     0  16520      0  0:00:01 --:--:--  0:00:01 16520
28961  100 28961    0     0  60335      0 --:--:-- --:--:-- --:--:-- 60209
Downloading https://raw.githubusercontent.com/elyra-ai/elyra/v3.8.1/etc/generic/requirements-elyra.txt
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
 1277  100  1277    0     0   2455      0 --:--:-- --:--:-- --:--:--  2460
Collecting packaging
  Downloading packaging-21.3-py3-none-any.whl (40 kB)
Collecting pyparsing!=3.0.5,>=2.0.2
  Downloading pyparsing-3.0.9-py3-none-any.whl (98 kB)
Installing collected packages: pyparsing, packaging
WARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv
Successfully installed packaging-21.3 pyparsing-3.0.9
WARNING: You are using pip version 21.2.4; however, version 22.0.4 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
[D 08:15:36.478] Parsing Arguments.....
[I 08:15:36.480] 'hello_world':'hello' - starting operation 
[I 08:15:36.480] 'hello_world':'hello' - Installing packages 
[I 08:15:36.480] Package not found. Installing ipykernel package with version 6.9.1...
[I 08:15:36.480] Package not found. Installing ipython package with version 8.0.1...
[I 08:15:36.480] Package not found. Installing ipython-genutils package with version 0.2.0...
[I 08:15:36.480] Package not found. Installing jinja2 package with version 3.0.3...
[I 08:15:36.480] Package not found. Installing jupyter-client package with version 7.1.2...
[I 08:15:36.480] Package not found. Installing jupyter-core package with version 4.9.2...
[I 08:15:36.481] Package not found. Installing MarkupSafe package with version 2.1.1...
[I 08:15:36.481] Package not found. Installing minio package with version 7.1.2...
[I 08:15:36.481] Package not found. Installing nbclient package with version 0.4.1...
[I 08:15:36.481] Package not found. Installing nbconvert package with version 6.4.5...
[I 08:15:36.481] Package not found. Installing nbformat package with version 5.1.3...
[I 08:15:36.481] Package not found. Installing papermill package with version 2.3.4...
[I 08:15:36.481] Package not found. Installing pyzmq package with version 19.0.1...
[I 08:15:36.481] Package not found. Installing prompt-toolkit package with version 3.0.5...
[I 08:15:36.481] Package not found. Installing requests package with version 2.25.1...
[I 08:15:36.481] Package not found. Installing tornado package with version 6.0.4...
[I 08:15:36.481] Package not found. Installing traitlets package with version 5.1.1...
[I 08:15:36.481] Package not found. Installing urllib3 package with version 1.26.5...
Collecting ipykernel==6.9.1
  Downloading ipykernel-6.9.1-py3-none-any.whl (128 kB)
Collecting ipython==8.0.1
  Downloading ipython-8.0.1-py3-none-any.whl (747 kB)
Collecting ipython-genutils==0.2.0
  Downloading ipython_genutils-0.2.0-py2.py3-none-any.whl (26 kB)
Collecting jinja2==3.0.3
  Downloading Jinja2-3.0.3-py3-none-any.whl (133 kB)
Collecting jupyter-client==7.1.2
  Downloading jupyter_client-7.1.2-py3-none-any.whl (130 kB)
Collecting jupyter-core==4.9.2
  Downloading jupyter_core-4.9.2-py3-none-any.whl (86 kB)
Collecting MarkupSafe==2.1.1
  Downloading MarkupSafe-2.1.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting minio==7.1.2
  Downloading minio-7.1.2-py3-none-any.whl (75 kB)
Collecting nbclient==0.4.1
  Downloading nbclient-0.4.1-py3-none-any.whl (65 kB)
Collecting nbconvert==6.4.5
  Downloading nbconvert-6.4.5-py3-none-any.whl (561 kB)
Collecting nbformat==5.1.3
  Downloading nbformat-5.1.3-py3-none-any.whl (178 kB)
Collecting papermill==2.3.4
  Downloading papermill-2.3.4-py3-none-any.whl (37 kB)
Collecting pyzmq==19.0.1
  Downloading pyzmq-19.0.1.tar.gz (1.2 MB)
Collecting prompt-toolkit==3.0.5
  Downloading prompt_toolkit-3.0.5-py3-none-any.whl (351 kB)
Collecting requests==2.25.1
  Downloading requests-2.25.1-py2.py3-none-any.whl (61 kB)
Collecting tornado==6.0.4
  Downloading tornado-6.0.4.tar.gz (496 kB)
Collecting traitlets==5.1.1
  Downloading traitlets-5.1.1-py3-none-any.whl (102 kB)
Collecting urllib3==1.26.5
  Downloading urllib3-1.26.5-py2.py3-none-any.whl (138 kB)
Collecting nest-asyncio
  Downloading nest_asyncio-1.5.5-py3-none-any.whl (5.2 kB)
Collecting debugpy<2.0,>=1.0.0
  Downloading debugpy-1.6.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (1.8 MB)
Collecting matplotlib-inline<0.2.0,>=0.1.0
  Downloading matplotlib_inline-0.1.3-py3-none-any.whl (8.2 kB)
Collecting pygments
  Downloading Pygments-2.12.0-py3-none-any.whl (1.1 MB)
Requirement already satisfied: setuptools>=18.5 in /usr/local/lib/python3.9/site-packages (from ipython==8.0.1) (58.1.0)
Collecting stack-data
  Downloading stack_data-0.2.0-py3-none-any.whl (21 kB)
Collecting backcall
  Downloading backcall-0.2.0-py2.py3-none-any.whl (11 kB)
Collecting pexpect>4.3
  Downloading pexpect-4.8.0-py2.py3-none-any.whl (59 kB)
Collecting jedi>=0.16
  Downloading jedi-0.18.1-py2.py3-none-any.whl (1.6 MB)
Collecting pickleshare
  Downloading pickleshare-0.7.5-py2.py3-none-any.whl (6.9 kB)
Collecting black
  Downloading black-22.3.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.5 MB)
Collecting decorator
  Downloading decorator-5.1.1-py3-none-any.whl (9.1 kB)
Requirement already satisfied: python-dateutil>=2.1 in /usr/local/lib/python3.9/site-packages (from jupyter-client==7.1.2) (2.8.2)
Collecting entrypoints
  Downloading entrypoints-0.4-py3-none-any.whl (5.3 kB)
Collecting certifi
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
Collecting async-generator
  Downloading async_generator-1.10-py3-none-any.whl (18 kB)
Collecting mistune<2,>=0.8.1
  Downloading mistune-0.8.4-py2.py3-none-any.whl (16 kB)
Collecting bleach
  Downloading bleach-5.0.0-py3-none-any.whl (160 kB)
Collecting beautifulsoup4
  Downloading beautifulsoup4-4.11.1-py3-none-any.whl (128 kB)
Collecting jupyterlab-pygments
  Downloading jupyterlab_pygments-0.2.2-py2.py3-none-any.whl (21 kB)
INFO: pip is looking at multiple versions of nbclient to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of minio to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of markupsafe to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of jupyter-core to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of jupyter-client to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of jinja2 to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of ipython-genutils to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of ipython to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of <Python from Requires-Python> to determine which version is compatible with other requirements. This could take a while.
INFO: pip is looking at multiple versions of ipykernel to determine which version is compatible with other requirements. This could take a while.
ERROR: Cannot install nbclient==0.4.1 and nbconvert==6.4.5 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested nbclient==0.4.1
    nbconvert 6.4.5 depends on nbclient<0.6.0 and >=0.5.0

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
WARNING: You are using pip version 21.2.4; however, version 22.0.4 is available.
You should consider upgrading via the '/usr/local/bin/python3 -m pip install --upgrade pip' command.
Traceback (most recent call last):
  File "/var/lib/pandas/jupyter-work-dir/bootstrapper.py", line 683, in <module>
    main()
  File "/var/lib/pandas/jupyter-work-dir/bootstrapper.py", line 666, in main
    OpUtil.package_install(user_volume_path=input_params.get("user-volume-path"))
  File "/var/lib/pandas/jupyter-work-dir/bootstrapper.py", line 547, in package_install
    subprocess.run([sys.executable, "-m", "pip", "install"] + to_install_list, check=True)
  File "/usr/local/lib/python3.9/subprocess.py", line 528, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['/usr/local/bin/python3', '-m', 'pip', 'install', 'ipykernel==6.9.1', 'ipython==8.0.1', 'ipython-genutils==0.2.0', 'jinja2==3.0.3', 'jupyter-client==7.1.2', 'jupyter-core==4.9.2', 'MarkupSafe==2.1.1', 'minio==7.1.2', 'nbclient==0.4.1', 'nbconvert==6.4.5', 'nbformat==5.1.3', 'papermill==2.3.4', 'pyzmq==19.0.1', 'prompt-toolkit==3.0.5', 'requests==2.25.1', 'tornado==6.0.4', 'traitlets==5.1.1', 'urllib3==1.26.5']' returned non-zero exit status 1.
```
</details>

**Expected behavior**
Successful run, with logs containing "hello world"

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 3.8.1
 - Installation source: PyPI
 - Deployment type: local installation (used [docs](https://elyra.readthedocs.io/en/stable/recipes/deploying-kubeflow-locally-for-dev.html) with a local `minikube` cluster instead of `Docker Desktop`)
 - Operating system: macos

**Comment by yarnabrina** [comment]
A related feature request: if a new system owned runtime image is added [here](https://github.com/elyra-ai/elyra/tree/master/etc/config/metadata/runtime-images) with the minimal requirements of `Elyra` (probably alpine/slim python + curl + ??), it'll be very helpful to demonstrate without using a ~1 GB Anaconda image or ~0.5 GB Pandas image, for both quickness and conciseness.

**Comment by ptitzler** [comment]
Thanks for reporting! We'll investigate. For the time being, can you use the `Tensorflow:2.8.0` image, which I've just confirmed still works.

**Comment by kevin-bates** [comment]
Looks like the `nbclient` dependency needs an update (to `nbclient==0.5.1`) to at least match what we use in our [elyra dependencies](https://github.com/elyra-ai/elyra/blob/8b9d93b5cfb6aedefb335de99b071925b38f5cca/setup.py#L84-L85).  We should also create a GH action job that creates a blank conda env and installs with [`requirements-elyra.txt`](https://github.com/elyra-ai/elyra/blob/master/etc/generic/requirements-elyra.txt), confirming the return code is zero so that this can be caught during builds.

**Comment by kevin-bates** [comment]
In building the GH action to run across the four python versions (3.7 -> 3.10), I've found that [we can't support images using Python 3.7](https://github.com/kevin-bates/elyra/runs/6390739760?check_suite_focus=true#step:3:128) since IPython 8.x is only available on Python >= 3.8.

Apparently none of our OOTB images run with Python 3.7.  We should probably make a support statement to this effect since we don't want to have to deal with different versions (relative to the Python version) within the requirements file.
